gno.land/cmd/gnoweb/static/js/renderer.js

```diff
@@ -6,3 +6,190 @@
 function renderUsernames(raw) {
   return raw.replace(/( |\n)@([_a-z0-9]{5,16})/, "$1[@$2](/r/demo/users:$2)");
 }
+
+function parseContent(source) {
+  marked.setOptions({ gfm: true });
+  const doc = new DOMParser().parseFromString(source, "text/html");
+  const contents = doc.documentElement.textContent;
+  return marked.parse(contents);
+}
+
+/*
+ *   ### ACCORDIONS ###
+ *
+ *   This content is licensed according to the W3C Software License at
+ *   https://www.w3.org/Consortium/Legal/2015/copyright-software-and-document
+ *
+ *   Desc: Simple accordion pattern example
+ */
+
+class Accordion {
+  constructor(domNode) {
+    this.buttonEl = domNode;
+    this.contentEl = this.buttonEl.nextElementSibling ?? this.buttonEl.parentElement.nextElementSibling;
+    this.open = this.buttonEl.getAttribute("aria-expanded") === "true";
+
+    this.buttonEl.addEventListener("click", this.onButtonClick.bind(this));
+  }
+
+  onButtonClick() {
+    this.toggle(!this.open);
+  }
+
+  toggle(open) {
+    if (open === this.open) {
+      return;
+    }
+
+    this.open = open;
+
+    this.buttonEl.setAttribute("aria-expanded", `${open}`);
+    if (open) {
+      this.contentEl.classList.remove("is-hidden");
+    } else {
+      this.contentEl.classList.add("is-hidden");
+    }
+  }
+}
+
+/*
+ *   ### TABS ###
+ *
+ *   This content is licensed according to the W3C Software License at
+ *   https://www.w3.org/Consortium/Legal/2015/copyright-software-and-document
+ *
+ *   Desc: Tablist widget that implements ARIA Authoring Practices
+ */
+
+class Tabs {
+  constructor(groupNode) {
+    this.tablistNode = groupNode;
+
+    this.tabs = [];
+
+    this.firstTab = null;
+    this.lastTab = null;
+
+    this.tabs = Array.from(this.tablistNode.querySelectorAll("[role=tab]"));
+    this.tabpanels = [];
+
+    for (var i = 0; i < this.tabs.length; i += 1) {
```

Comment:
`this.tabs.forEach`? or `for (let tab of this.tabs)`

Reply:
The first one if you like functional programming and the second one for readability, mine for speed.

Reply:
Let use the `for of`